google/cloud/aiplatform/datasets/tabular_dataset.py

```diff
@@ -40,7 +42,7 @@ class TabularDataset(datasets._Dataset):
     )
 
     @property
-    def column_names(self) -> List[str]:
```

Comment:
The set comparison is trivial because col_names are unique and order is not a factor.

List implies an order which is not relevant for column names and makes the unit tests a tiny (very tiny) bit more complicated to write.

Reply:
I see your point about a breaking change. What do you recommend?

Reply:
I agree with the motivation to change the return type to a Set. Let's do the following:

1. Leave the return type as List to avoid the breaking change.
2. Proceed with this PR mainly as is.
3. Open a ticket to track the return type change to Set.
4. Tentatively plan to implement the return type change when we get closer to a larger breaking change and major version rev.

Reply:
Agreed on all points.

Reply:
https://b.corp.google.com/issues/193044977

Reply:
Tracked in b/193044977

Reply:
I kept the private method return types as Set as I assume that it's acceptable to break private methods.

Reply:
@sasha-gitg made the changes.